src/controllers/dynakube/oneagent/daemonset/daemonset.go

```diff
@@ -254,16 +254,16 @@ func (dsInfo *builderInfo) imagePullSecrets() []corev1.LocalObjectReference {
 }
 
 func (dsInfo *builderInfo) securityContext() *corev1.SecurityContext {
-	if dsInfo.instance.IsOneAgentPrivileged() {
-		runAsPrivileged := true
-		securityContext := &corev1.SecurityContext{
-			Privileged: &runAsPrivileged,
-		}
-		return securityContext
+	securityContext := &corev1.SecurityContext{
+		RunAsNonRoot: address.Of(true),
+		RunAsUser:    address.Of(int64(1000)),
+		RunAsGroup:   address.Of(int64(1000)),
```

Comment:
question: shouldn't it be the same as what we set for the operator, just to be consistent, 
or there is a reason for this user/group?

Reply:
I reused the user/group we previously only used in case of read only mode
since it's a different image than the operator image, the operator user wouldn't be available